### PR TITLE
PLANNER-973 Improve UI for the advanced annotation editor

### PR DESCRIPTION
**Pull request by yurloc**
See https://issues.jboss.org/browse/PLANNER-973 for screenshots.

### CODE REVIEW

**Comment by yurloc** [comment]
Jenkins build failed due to a JVM crash
```
11:54:49 #
11:54:49 # A fatal error has been detected by the Java Runtime Environment:
11:54:49 #
11:54:49 #  SIGSEGV (0xb) at pc=0x00007faddc316587, pid=10728, tid=0x00007fad8d6d2700
11:54:49 #
11:54:49 # JRE version: Java(TM) SE Runtime Environment (8.0_152-b16) (build 1.8.0_152-b16)
11:54:49 # Java VM: Java HotSpot(TM) 64-Bit Server VM (25.152-b16 mixed mode linux-amd64 compressed oops)
11:54:49 # Problematic frame:
11:54:49 # V  [libjvm.so+0x43c587][INFO] 
11:54:49   ciObjectFactory::create_new_metadata(Metadata*)+0x327
11:54:49 #
11:54:49 # Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
11:54:49 #
11:54:49 # An error report file with more information is saved as:
11:54:49 # /home/jenkins/workspace/kie-wb-common-pullrequests/upstream-repos/droolsjbpm-tools/hs_err_pid10728.log
11:54:49 #
11:54:49 # Compiler replay data is saved as:
11:54:49 # /home/jenkins/workspace/kie-wb-common-pullrequests/upstream-repos/droolsjbpm-tools/replay_pid10728.log
11:54:49 #
11:54:49 # If you would like to submit a bug report, please visit:
11:54:49 #   http://bugreport.java.com/bugreport/crash.jsp
11:54:49 #
```
somewhere near
```
11:54:49 [INFO] --- tycho-packaging-plugin:1.0.0:package-feature (default-package-feature) @ org.kie.eclipse.navigator.feature ---
11:54:49 [INFO] Building jar: /home/jenkins/workspace/kie-wb-common-pullrequests/upstream-repos/droolsjbpm-tools/drools-eclipse/org.kie.eclipse.navigator.feature/target/org.kie.eclipse.navigator.feature-7.6.0-SNAPSHOT.jar
11:54:49 [INFO] 
11:54:49 [INFO] --- tycho-p2-plugin:1.0.0:p2-metadata-default (default-p2-metadata-default) @ org.kie.eclipse.navigator.feature ---
```

**Comment by yurloc** [comment]
Jenkins retest this.

**Comment by yurloc** [comment]
Asking @rsynek for QE review and @wmedvede for peer review, please.

**Comment by rsynek** [comment]
The failure is not connected with this change, overall looks good.

**Comment by yurloc** [comment]
Unstable test:
```
java.lang.AssertionError: Number of docs fulfilling the given query criteria expected:<3> but was:<4>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.kie.workbench.common.services.refactoring.backend.server.IndexingTest.searchFor(IndexingTest.java:240)
	at org.kie.workbench.common.services.refactoring.backend.server.drl.IndexDrlLHSTypeTest.testIndexDrlLHSType(IndexDrlLHSTypeTest.java:64)
```

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/drl/IndexDrlLHSTypeTest.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public void testIndexDrlLHSType() throws IOException, InterruptedException {
                            drl3 );
         ioService().endBatch();
 
-        Thread.sleep( 12000 ); //wait for events to be consumed from jgit -> (notify changes -> watcher -> index) -> lucene index
+        Thread.sleep( 15000 ); //wait for events to be consumed from jgit -> (notify changes -> watcher -> index) -> lucene index
```
Comment (jhrcek) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/drl/IndexDrlLHSTypeTest.java]: @yurloc based on my [long term research](http://janhrcek.cz/random-failures/#/class/org.kie.workbench.common.services.refactoring.backend.server.drl.IndexDrlLHSTypeTest/method/testIndexDrlLHSType) into this I can tell you that this timeout increase won't have any effect on test stability. There's some concurrency bug in the indexing backend which causes these indexing test failures and increasing timeout will just make the tests run longer..

Comment (rsynek) [reply kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/drl/IndexDrlLHSTypeTest.java]: If the test is unstable, I would rather suggest marking it `@Ignored` for this moment and filing an issue to stabilize it (or to reveal and fix the issue Jan is talking about).

Comment (jhrcek) [reply kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/drl/IndexDrlLHSTypeTest.java]: @rsynek there's probably just one person who has enough knowledge of the codebase to be able to fix it. It's @porcelli

Comment (yurloc) [reply kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/drl/IndexDrlLHSTypeTest.java]: @jhrcek Ok, I won't touch the sleep amount in this PR.

**Comment by yurloc** [comment]
Hi @manstis, can you please take a look at this and merge if it's OK?

**Comment by manstis** [comment]
@wmedvede One for you I believe?

**Comment by wmedvede** [comment]
@manstis sorry for the delay guys, I'll review tomorrow early. thanks.

**Comment by yurloc** [comment]
Thanks @wmedvede.
@manstis, please go ahead and merge (I suppose you're the gatekeeper).

**Comment by manstis** [comment]
@yurloc Yes one of them (Roger and Pere are too, IIRC)

@jhrcek Could you please approve before merge (since you posed a question).

**Comment by yurloc** [comment]
@manstis mergeable 😉

**Pull request merged**